service/history/transferQueueProcessorBase.go

```diff
@@ -212,6 +215,26 @@ func (t *transferQueueProcessorBase) recordWorkflowClosed(
 		return nil
 	}
 
+	// publish to kafka
+	if t.visibilityProducer != nil {
+		msgType := indexer.VisibilityMsgTypeClosed
+		msg := &indexer.VisibilityMsg{
+			MsgType:       &msgType,
+			DomainID:      common.StringPtr(domainID),
+			WorkflowID:    common.StringPtr(wid),
+			RunID:         common.StringPtr(execution.GetRunId()),
+			WorkflowType:  common.StringPtr(workflowTypeName),
+			StartTime:     common.Int64Ptr(startTimeUnixNano),
+			CloseTime:     common.Int64Ptr(endTimeUnixNano),
+			CloseStatus:   &closeStatus,
+			HistoryLength: common.Int64Ptr(historyLength),
+		}
+		err := t.visibilityProducer.Publish(msg)
```

Comment:
Can you make sure we have request/failure/latency metric emitted for visibilityProducer.

Reply:
Added. Changed to use metricsProducer